OSDOCS#9971:Added breakglass feature modules for HCP clusters (#73863)
openshift/openshift-docs: modules/rosa-create-objects.adoc

$ rosa create cluster --cluster-name=mycluster -region=us-east-1 --enable-autoscaling --min-replicas=2 --max-replicas=5
----

[id="rosa-create-break-glass-credential_{context}"]
== create break glass credential

Create a break glass credential for a hosted control plane cluster with external authentication enabled.

.Syntax
[source,terminal]
----
$ rosa create break-glass-credential --cluster=<cluster_name> [arguments]
----

.Arguments
[cols="30,70"]
|===
|Option |Definition

|--cluster <cluster_name>\|<cluster_id>
|Required. The name or ID of the cluster to which the break glass credential will be added.

| --expiration duration

> |Optional. How long a break glass credential can be used before expiring. The expiration duration must be a minimum of 10 minutes and a maximum of 24 hours. If you do not enter a value, the expiration duration defaults to 24 hours.

Reviewer: ```suggestion
|Optional: How long a break glass credential can be used before expiring. The expiration duration must be a minimum of 10 minutes and a maximum of 24 hours. If you do not enter a value, the expiration duration defaults to 24 hours.
```

|--username
|Optional. The username for the break glass credential. If you do not enter a value, the username defaults to the name of the user creating the break glass credential.
|===

.Optional arguments inherited from parent commands
[cols="30,70"]
|===
|Option |Definition

|--help
|Shows help for this command.

|--debug
|Enables debug mode.

|--interactive
|Enables interactive mode.

|--profile
|Specifies an AWS profile (string) from your credentials file.

|--region
|Specifies an AWS region, overriding the `AWS_REGION` environment variable.

|--yes
|Automatically answers `yes` to confirm the operation.
|===

.Examples
Add a break glass credential to a cluster named `mycluster`.

.Syntax
[source,terminal]
----

> $ rosa break-glass-credential --cluster-name=mycluster

Reviewer 2: Here the command must be "rosa create break-glass-credential --cluster=mycluster"

$ rosa create cluster --cluster-name=mycluster -region=us-east-1 --enable-autoscaling --min-replicas=2 --max-replicas=5
----

[id="rosa-create-break-glass-credential_{context}"]
== create break glass credential

Create a break glass credential for a hosted control plane cluster with external authentication enabled.

.Syntax
[source,terminal]
----
$ rosa create break-glass-credential --cluster=<cluster_name> [arguments]
----

.Arguments
[cols="30,70"]
|===
|Option |Definition

|--cluster <cluster_name>\|<cluster_id>
|Required. The name or ID of the cluster to which the break glass credential will be added.

> | --expiration duration

Reviewer 3: Is it required to add "duration" here? It means it is duration type

|Optional: How long a break glass credential can be used before expiring. The expiration duration must be a minimum of 10 minutes and a maximum of 24 hours. If you do not enter a value, the expiration duration defaults to 24 hours.

|--username

> |Optional. The username for the break glass credential. If you do not enter a value, the username defaults to the name of the user creating the break glass credential.

Reviewer 2: ```suggestion
|Optional. The username for the break glass credential. If you do not enter a value, random username will be generated.
```

|===

.Optional arguments inherited from parent commands
[cols="30,70"]
|===
|Option |Definition

|--help
|Shows help for this command.

|--debug
|Enables debug mode.

|--interactive
|Enables interactive mode.

|--profile
|Specifies an AWS profile (string) from your credentials file.

|--region
|Specifies an AWS region, overriding the `AWS_REGION` environment variable.

|--yes
|Automatically answers `yes` to confirm the operation.
|===

.Examples
Add a break glass credential to a cluster named `mycluster`.

.Syntax
[source,terminal]
----

> $ rosa break-glass-credential --cluster=mycluster

Reviewer 2: The command must be "rosa create break-glass-credential --cluster=mycluster"

Author: Sorry- that's what happens when I try to do stuff before coffee!

----

Add a break glass credential to a cluster named `mycluster` using the interactive mode.

.Syntax
[source,terminal]
----

> $ rosa break-glass-credential --cluster=mycluster -i

Reviewer 2: The command must be "rosa create break-glass-credential --cluster=mycluster -i"
